pygmt/src/coast.py

```diff
@@ -183,10 +188,26 @@ def coast(self, **kwargs):
     {V}
     """
     kwargs = self._preprocess(**kwargs)  # pylint: disable=protected-access
+
+    if "clip" in kwargs:
+        if kwargs["clip"] in ["land", "water", "end"]:
+            if kwargs["clip"] == "land":
+                kwargs["G"] = True
+            elif kwargs["clip"] == "water":
+                kwargs["S"] = True
+            elif kwargs["clip"] == "end":
+                kwargs["Q"] = True
+            kwargs.pop("clip")
+        else:
+            raise GMTInvalidInput(
+                "Invalid clip parameter. Must be one of 'land', 'water', and 'end'."
+            )
```

Comment:
Please also add a test to make sure `GMTInvalidInput` is raised when invalid parameter like `clip="invalid"` is given.

Reply:
Done!